src/iop/highlights.c

```diff
@@ -1916,7 +1918,11 @@ static void process_visualize(dt_dev_pixelpipe_iop_t *piece, const void *const i
 
   const float *const in = (const float *const)ivoid;
   float *const out = (float *const)ovoid;
-  const float clip = data->clip;
+
+  /* As some of the internal algorithms use a smaller value for clipping than given by the UI
+     the visualizing is wrong for those algos. It seems to be a a minor issue but sometimes significant. */
+  const float clip_magics[6] = { 1.0f, 1.0f, 0.987f, 0.995f, 0.987f, 0.987f };  
```

Comment:
This and the definition in process_cl should be made global (and static) so that there is only a single definition.  Otherwise, they will tend to desync when new recovery algorithms are added.

Reply:
good point :-)